dashboard/src/components/AppView/AppControls.tsx

```diff
@@ -10,35 +10,52 @@ interface IAppControlsProps {
 }
 
 interface IAppControlsState {
+  migrate: boolean;
   modalIsOpen: boolean;
   redirectToAppList: boolean;
   upgrade: boolean;
+  deleting: boolean;
 }
 
 class AppControls extends React.Component<IAppControlsProps, IAppControlsState> {
   public state: IAppControlsState = {
+    deleting: false,
+    migrate: false,
     modalIsOpen: false,
     redirectToAppList: false,
     upgrade: false,
   };
 
   public render() {
     const { name, namespace } = this.props.app.data;
+    if (this.props.app.hr && this.props.app.hr.metadata) {
+      return (
+        <div className="AppControls">
+          <button className="button" onClick={this.handleUpgradeClick}>
+            Upgrade
+          </button>
+          {this.state.upgrade && <Redirect push={true} to={`/apps/ns/${namespace}/edit/${name}`} />}
+          <button className="button button-danger" onClick={this.openModel}>
+            Delete
+          </button>
+          <ConfirmDialog
+            onConfirm={this.handleDeleteClick}
+            modalIsOpen={this.state.modalIsOpen}
+            loading={this.state.deleting}
+            closeModal={this.closeModal}
+          />
+          {this.state.redirectToAppList && <Redirect to={`/apps/ns/${namespace}`} />}
+        </div>
+      );
+    }
     return (
       <div className="AppControls">
-        <button className="button" onClick={this.handleUpgradeClick}>
-          Upgrade
-        </button>
-        {this.state.upgrade && <Redirect push={true} to={`/apps/ns/${namespace}/edit/${name}`} />}
-        <button className="button button-danger" onClick={this.openModel}>
-          Delete
+        <button className="button" onClick={this.handleMigrateClick}>
+          Migrate
```

Comment:
should we call this button `Import App`?

Reply:
I set this back to `Migrate` since there is now a banner that explains it in detail